addons/pos_gift_card/static/src/js/models.js

```diff
@@ -0,0 +1,55 @@
+odoo.define("pos_gift_card.gift_card", function (require) {
+  "use strict";
+
+  const { Order, Orderline } = require("point_of_sale.models");
+  const Registries = require('point_of_sale.Registries');
+
+
+  const PosGiftCardOrder = (Order) => class PosGiftCardOrder extends Order {
+    //@override
+    _reduce_total_discount_callback(sum, orderLine) {
+        if (this.pos.config.gift_card_product_id[0] === orderLine.product.id) {
+            return sum;
+        }
+        return super._reduce_total_discount_callback(...arguments);
+    }
+    //@override
+    set_orderline_options(orderline, options) {
+      super.set_orderline_options(...arguments);
+      if (options && options.generated_gift_card_ids) {
+        orderline.generated_gift_card_ids = [options.generated_gift_card_ids];
+      }
+      if (options && options.gift_card_id) {
+        orderline.gift_card_id = options.gift_card_id;
+      }
+    }
+    //override
+    wait_for_push_order() {
```

Comment:
I can't remember this in the original PR. Or should this belong to a different commit?

Reply:
Oh, this file shouldn't exist. This change should be in pos_loyalty.